docs/docs/dev_docs/contracts/storage.md

```diff
@@ -4,10 +4,16 @@
 
 State variables must be declared inside a struct. (This enables us to declare types composed of nested generics in Noir - see [types](./types.md)).
 
-By way of example, we could define a private state variable `balances`, mapping user addresses to their token balances:
+We could define any kinds of state variables in the Storage struct:
 
-#include_code storage-declaration /yarn-project/noir-contracts/src/contracts/private_token_contract/src/storage.nr rust
+#include_code storage-declaration /yarn-project/noir-contracts/src/contracts/docs_example_contract/src/storage.nr rust
```

Comment:
I'll start reviewing this now (in between other meetings). It's looking lovely!

My first comment: If the example contract is going to be used throughout the syntax explanations (which is a very nice idea), should we have a page or paragraph which introduces the _purpose_ of this contract? What is the game? What is the objective of the game? What are the rules? What do players do?

It feels like this explanation could be both a README within the contract's directory, but also as a dedicated page or paragraph of these docs?